app/src/features/surveys/view/SurveyMap.tsx

```diff
@@ -41,15 +82,71 @@ const SurveyMap = (props: ISurveyMapProps) => {
           <FullScreenScrollingEventHandler bounds={bounds} scrollWheelZoom={false} />
           <SetMapBounds bounds={bounds} />
 
-          {props.mapPoints?.map((nonEditableGeo: INonEditableGeometries, index: number) => {
-            const key = nonEditableGeo.feature.id ?? index;
+          {props.mapPoints?.map((mapPoint: ISurveyMapPoint, index: number) => {
+            const { key } = mapPoint;
+            const isLoading = !mapPointMetadata[key];
 
             return (
               <GeoJSON
                 key={key}
-                data={nonEditableGeo.feature}
-                pointToLayer={(_, latlng) => coloredPoint({ latlng, fillColor: '#1f7dff', borderColor: '#ffffff' })}>
-                {nonEditableGeo.popupComponent}
+                data={mapPoint.feature}
+                pointToLayer={(_, latlng) => coloredPoint({ latlng, fillColor: '#1f7dff', borderColor: '#ffffff' })}
+                onEachFeature={(feature, layer) => {
+                  layer.on({
+                    popupopen: () => {
```

Comment:
Does this need to check `mapPointMetadata` for previously loaded metadata for the point, and only call onLoad if it is missing?

Reply:
Good catch! It should not make repeated fetches.